src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs

```diff
@@ -34,6 +34,18 @@ public class AvoidAnonymousTypes : TypedRule<SwaggerObject>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
```

Comment:
I think this is more of an AutoRest recommendation, so that we know the name of the class and don't make one up, I would probably do ARM and Data Plane, if we wanted to leave default out.